src/Illuminate/Queue/BeanstalkdQueue.php

```diff
@@ -117,7 +118,11 @@ public function pop($queue = null)
     {
         $queue = $this->getQueue($queue);
 
-        $job = $this->pheanstalk->watchOnly($queue)->reserve(0);
+        $this->pheanstalk->watchOnly($queue);
```

Comment:
`watchOnly()` return `$this` so this should be fine. For 5.8 we can use the existing code.

```php
$job = $this->pheanstalk->watchOnly($queue)->reserveWithTimeout($this->blockFor);
```